espnet2/s2st/synthesizer/discrete_synthesizer.py

```diff
@@ -105,7 +103,7 @@ def __init__(
             dec_idim = idim + spk_embed_dim
         elif self.spk_embed_integration_type == "add":
             dec_idim = idim
-            self.projection = torch.nn.Linear(self.spk_embed_dim, encoder_output_size)
+            self.projection = torch.nn.Linear(self.spk_embed_dim, dec_idim)
```

Comment:
is it OK?
@ftshijt, can you confirm it?

Reply:
There is no code line initially assigning encoder_output_size, So I inferred the corresponding variable based on the TransformerDecoder output size

Reply:
Confirmed, this change should be fine! Thanks!